microprofile/WFLY-17138-convert-mp-metrics-subsystem-to-legacy-subsystem.adoc

```diff
@@ -0,0 +1,104 @@
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [X] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+MicroProfile Metrics subsystem will be removed from standard configs.
+
+A `MigrateOperation` will be added to remove the extension from existing server configs upon server startup (in admin-only mode).
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+While Micrometer is replacing MicroProfile Metrics in WildFly, it is a non-requirement to install the new extension/subsystem as part of the `MigrateOperation`. WildFly administrators will need to explicitly choose between Micrometer and the existing (and similar to MP Metrics) "base" metrics extension.
+
+== Backwards Compatibility
+
+Since this represents the removal of a subsystem, any deployment dependent on the APIs provided will no longer deploy or run.
+
+All other subsystems, except for those with optional integrations (see below) should be unaffected by this removal. Those subsystems will see a reduction in functionality commensurate with the removal of MicroProfile Metrics.
+
+=== Default Configuration
+
+The default configuration for MicroProfile configurations will be altered to remove the MicroProfile Metrics subsystem. Likewise, the relevant Galleon layers will also have the extension removed from their definitions.
+
+=== Importing Existing Configuration
+
+With the change of the subsystem to admin-only, existing configurations will fail to start in normal standalone modes. Such configurations will need to be started in admin mode to allow for the altering of the configuration, or the extension and subsystem will need to be removed manually.
+
+=== Deployments
+
+Deployments that use MicroProfile Metrics APIs directly will fail to deploy. These applications will need to modified to remove this dependency.
+
+Applications that do not use the affected API directly should deploy and run without modification or issue.
+
+=== Interoperability
+
+Two other MicroProfile specs have optional integrations with Metrics, those being Fault Tolerance and Reactive Messaging. The WildFly extensions for these two specifications have been modified so as not to leverage this integration, allowing applications using those specs to continue to deploy and run. They will not, however, generate metrics data, so if the deployment environment is expecting this data, adjustments by the administrator will need to be made.
```

Comment:
s/adjustments by the administrator/adjustments by administrators or developers/ since I am thinking about applications that do use the MP FT APIs with integrated metrics. WDYT?

Reply:
Makes sense. Done.